compiler/src/Language/Bond/Codegen/Java/SerializationMethods.hs

```diff
@@ -278,3 +280,13 @@ readMap java keyType valueType fieldName depth =
         countLocal = "count" ++ show depth
         keyLocal = "k" ++ show depth
         valueLocal = "v" ++ show depth
+
+readBonded :: MappingContext -> Type -> String -> Text
+readBonded java bondedType fieldName =
```

Comment:
`bondedType` matches any type, not just `BT_Bonded`. We've got a patten that we often use in cases like this:

```
readBonded java bondedType@(BT_Bonded _) fieldName = ...
readBonded _ _ _ = error "Impossible..."
```